chatkit-core/src/test/kotlin/com/pusher/chatkit/UserEnrichmentErrorFunctionalTest.kt

```diff
@@ -0,0 +1,68 @@
+package com.pusher.chatkit
+
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.users.UserSubscriptionEvent
+import com.pusher.chatkit.users.usersFetchFailingWith
+import com.pusher.util.Result
+import elements.Error
+import elements.Errors
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.util.concurrent.BlockingQueue
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.TimeUnit
+
+object UserEnrichmentErrorFunctionalTest : Spek({
+
+    val initialState = UserSubscriptionEvent.InitialState(
+            simpleUser("alice"),
+            listOf(simpleRoom("roomId1", "Room 1")),
+            listOf(RoomReadStateApiType("roomId1", 0, null)),
+            listOf(RoomMembershipApiType("roomId1", listOf("alice")))
+    )
+    val networkError = Errors.network("test error")
+
+    describe("given error when fetching user joining the room") {
```

Comment:
Not sure I understand `given error when fetching user joining the room` - does this mean when a user joins the room, we try to get more information about them and that part errors?

Reply:
Yep that's what it means :) It's also clear from the lines just below the given part.